File: VB/LongRunningOperations.Module/Controllers/StartProcessingController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.ExpressApp.ConditionalAppearance
Imports DevExpress.Persistent.Base
Imports LongRunningOperations.Module.BusinessObjects

Namespace LongRunningOperations.Module.Controllers
	Public Class StartProcessingController
		Inherits ViewController
		Public Sub New()
			Dim action As New SimpleAction(Me, "StartProcessing", PredefinedCategory.Edit)
			AddHandler action.Execute, AddressOf action_Execute
		End Sub
		Private Sub action_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs)
			StartProcessing()
		End Sub
		Public Sub StartProcessing()
			Dim os As IObjectSpace = Application.CreateObjectSpace()
			Dim obj As ObjectForLongRunningOperations = os.CreateObject(Of ObjectForLongRunningOperations)()
			obj.ObjectToProcess = CType(os.GetObject(View.CurrentObject), Object1)
			If obj.ObjectToProcess Is Nothing Then
				Throw New ArgumentException()
			End If
			os.CommitChanges()
			CType(View.CurrentObject, Object1).SetIsInProcessing(True)
			Frame.GetController(Of AppearanceController)().Refresh()
		End Sub
	End Class
End Namespace

File: CS/LongRunningOperations.Module/Controllers/StartProcessingController.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.ExpressApp.ConditionalAppearance;
using DevExpress.Persistent.Base;
using LongRunningOperations.Module.BusinessObjects;

namespace LongRunningOperations.Module.Controllers {
    public class StartProcessingController : ViewController {
        public StartProcessingController() {
            SimpleAction action = new SimpleAction(this, "StartProcessing", PredefinedCategory.Edit);
            action.Execute += new SimpleActionExecuteEventHandler(action_Execute);
        }
        void action_Execute(object sender, SimpleActionExecuteEventArgs e) {
            StartProcessing();
        }
        public void StartProcessing() {
            IObjectSpace os = Application.CreateObjectSpace();
            ObjectForLongRunningOperations obj = os.CreateObject<ObjectForLongRunningOperations>();
            obj.ObjectToProcess = (Object1)os.GetObject(View.CurrentObject);
            if(obj.ObjectToProcess == null) {
                throw new ArgumentException();
            }
            os.CommitChanges();
            ((Object1)View.CurrentObject).SetIsInProcessing(true);
            Frame.GetController<AppearanceController>().Refresh();
        }
    }
}
